storm-webapp/src/main/java/org/apache/storm/daemon/logviewer/LogviewerServer.java

```diff
@@ -52,7 +55,7 @@
 /**
  * The main entry of Logviewer.
  */
-public class LogviewerServer implements AutoCloseable {
+public class LogviewerServer implements AutoCloseable, MetricSet {
```

Comment:
It would be weirder since `getMetrics` is an instance method. So I have to do `[SOME_ENUM].getMetrics()`

Reply:
You're right, but can't a lambda solve this? Would it work to do `registerMetricSet(ExceptionMeters::getMetrics)` in line 132 instead?

Reply:
No it would be `registerMetricSet(ExceptionMeters.[SOME_ENUM])` or 

```java
enum ExceptionMeters implements MetricSet {
    ENUM1;

    private static final Map<...> map = ...;

    public static MetricSet getMetricSet() {
        return ENUM1;

    @Override
    public Map<...> getMetrics() {
        return map;
    }
}
```

Reply:
I can think of an Alternative

```java
enum ExceptionMeters {
    ENUM1;

    private static final Map<...> map = ...;
    private static final MetricSet set = () -> map;

    public static MetricSet getMetricSet() {
        return set;
}
```

Reply:
`StormMetricsRegistry.registerMetricSet(ExceptionMeters::getMetrics);` works fine for me?

Reply:
Just to be clear, I'm talking about removing `implements MetricSet` from LogviewerServer and replacing it with `StormMetricsRegistry.registerMetricSet(ExceptionMeters::getMetrics);`. ExceptionMeters doesn't need to implement MetricSet.

Reply:
Ah I see. That's brilliant. Thanks